### PR TITLE
Add coverage for activation key host collection disassociation

### LINKED ISSUE #3440: Create coverage for host collection disassociation from activation key
Refer to #3434 for more details

### PR DESCRIPTION
**Pull request by oshtaier**
Closes #3440

```
nosetests tests/foreman/api/test_activationkey.py -m test_positive_remove_host_collection
.
----------------------------------------------------------------------
Ran 1 test in 19.851s

OK
```

Depends on https://github.com/SatelliteQE/nailgun/pull/299

### CODE REVIEW

File: tests/foreman/api/test_activationkey.py [hunk 1]
```diff
@@ -302,6 +302,40 @@ def test_positive_add_host_collections(self):
...
+    def test_positive_remove_host_collection(self):
+        """Disassociate host collection from the activation key
+
+        @Assert:
+
+        1. By default, an activation key is associated with no host
+           collections.
+        2. Associating host collection with activation key add it to the list.
+        3. Disassociating host collection from the activation key actually
+        removes it from the list
```
Comment (abalakh) [inline tests/foreman/api/test_activationkey.py]:
```python
/home/travis/build/SatelliteQE/robottelo/tests/foreman/api/test_activationkey.py:docstring of tests.foreman.api.test_activationkey.ActivationKeyTestCase.test_positive_remove_host_collection:8: WARNING: Enumerated list ends without a blank line; unexpected unindent.
```

**Comment by abalakh** [comment]
ACK pending docstring fix

**Comment by oshtaier** [comment]
Done

**Comment by rochacbruno** [comment]
ACK

**Pull request merged**